geomesa-convert/geomesa-convert-osm/src/main/scala/org/locationtech/geomesa/convert/osm/OsmWaysConverter.scala

```diff
@@ -0,0 +1,146 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.convert.osm
+
+import java.io.InputStream
+import java.nio.file.Files
+import java.sql.DriverManager
+
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import com.vividsolutions.jts.geom.Coordinate
+import de.topobyte.osm4j.core.model.iface._
+import de.topobyte.osm4j.core.model.impl.Node
+import de.topobyte.osm4j.pbf.seq.PbfIterator
+import de.topobyte.osm4j.xml.dynsax.OsmXmlIterator
+import org.geotools.geometry.jts.JTSFactoryFinder
+import org.locationtech.geomesa.convert.Transformers.{EvaluationContext, Expr}
+import org.locationtech.geomesa.convert._
+import org.locationtech.geomesa.utils.classpath.PathUtils
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.immutable.IndexedSeq
+
+class OsmWaysConverter(val targetSFT: SimpleFeatureType,
+                       val idBuilder: Expr,
+                       val inputFields: IndexedSeq[Field],
+                       val userDataBuilder: Map[String, Expr],
+                       val validating: Boolean,
+                       val pbf: Boolean,
+                       val needsMetadata: Boolean) extends ToSimpleFeatureConverter[OsmWay] with LazyLogging {
+
+  private def gf = JTSFactoryFinder.getGeometryFactory
+  private val toArray = if (needsMetadata) { OsmField.toArrayWithMetadata _ } else { OsmField.toArrayNoMetadata _ }
+
+  // create a temporary h2 database to store our nodes
+  // the ways only refer to nodes by reference, but we can't keep the whole file in memory
+  private val h2Dir = Files.createTempDirectory("geomesa-convert-h2").toFile
+  private val connection = {
+    Class.forName("org.h2.Driver")
```

Comment:
yeah, this is the typical pattern used for jdbc to ensure that the classpath is correct I guess